platform/os_linux.go

```diff
@@ -81,7 +82,13 @@ func (p *execClient) ExecuteCommand(command string) (string, error) {
 
 	var stderr bytes.Buffer
 	var out bytes.Buffer
-	cmd := exec.Command("sh", "-c", command)
+
+	// Create a new context and add a timeout to it
+	const timeoutSeconds = 10
```

Comment:
how about we defined this in execClient object so that we configure it for unit test as well

Reply:
I like the idea. Moved timeout to execclient with both default option and an option to override default.